### PR TITLE
Bootstrap 5 study edit/create

### PR DESCRIPTION
**Pull request by becky-gilbert**
Initial attempt at updating the study edit view to use bootstrap 5. 
Browser is zoomed out a bit in screenshots because the form is so long.
I still need to check the form validation on this. I'll request reviews when it's ready!

study_edit_1
study_edit_2
study_edit_3
study_edit_4
study_edit_5
study_edit_6

### CODE REVIEW

**Comment by becky-gilbert** [comment]
Ok I think the form validation is working. There are some things that we could improve, but it's working the same way as on staging.

https://user-images.githubusercontent.com/9041788/221955074-4ba2c2fc-53be-4875-863f-eebc433f8ab4.mov

https://user-images.githubusercontent.com/9041788/221955444-69321252-8b6a-4581-a052-7da72a74c947.mov

Screenshot 2023-02-28 at 11 03 11 AM

Screenshot 2023-02-28 at 11 04 10 AM

Screenshot 2023-02-28 at 11 06 05 AM

Screenshot 2023-02-28 at 11 07 28 AM

**Comment by becky-gilbert** [comment]
Sorry, not ready for review quite yet.

**Comment by becky-gilbert** [comment]
I'm adding the study create view onto this pull request since I've just made some changes to study edit that are shared with create, and thus will affect that view as well.

**Comment by becky-gilbert** [comment]
@okaycj as discussed, I pulled the JS and CSS out of the HTML files and put then in separate files (on per HTML file). But I didn't get rid of`$(document).ready` in a few of the JS files because there were some bits of JS that needed to run immediately and others that needed to run after the page loads (and thus I couldn't use `defer`). 

This is almost ready for review. I need to push my changes to the study create view, and maybe clean up the commits/diff to make it easier to review (rebase or merge `bootstrap-5`?).

File: studies/templates/studies/study_edit.html [hunk 1]
```diff
@@ -39,99 +18,96 @@
...
             {% if message.extra_tags != 'user_added' and message.extra_tags != 'user_removed' %}
                 <div id="details" class="alert {{ message.tags }} alert-dismissable">
-                    <a href="#" class="close" data-dismiss="alert" aria-label="close">×</a>
+                    <a href="#" class="close" data-bs-dismiss="alert" aria-label="close">×</a>
                     {{ message }}
                 </div>
             {% endif %}
         {% endfor %}
     {% endif %}
 {% endblock flash %}
```
Comment (okaycj) [inline studies/templates/studies/study_edit.html:27]: I am inclined to remove the messages block. Did you find it to be necessary in this form?

**Review by okaycj — APPROVED** [review]
This is very complex form and you did a great job wrangling it.  Just a quick question but other wise ready to go!  Thank you.

File: studies/templates/studies/study_edit.html (hunk 1 above)
Comment (becky-gilbert) [reply studies/templates/studies/study_edit.html:27]: Ah good question. I believe the messages block is used after a study is saved successfully. 


And the change here (`data-bs-dismiss`) adds the close (X) button on the right side, and apparently some functionality: https://getbootstrap.com/docs/5.0/components/alerts/#dismissing

Comment (okaycj) [reply studies/templates/studies/study_edit.html:27]: There's a message tag in the base template.  Does that not do the same thing?

Comment (becky-gilbert) [reply studies/templates/studies/study_edit.html:27]: Ahh sorry, missed that. I'll check now.

Comment (becky-gilbert) [reply studies/templates/studies/study_edit.html:27]: Yep you're right, I'll take it out. 
Do I need to replace it with `{% bootstrap_messages %}` as in here:
https://github.com/lookit/lookit-api/blob/f7cbb535bc1b95346403378e47d0c8600659456b/studies/templates/studies/study_form.html#L29
and here:
https://github.com/lookit/lookit-api/blob/f7cbb535bc1b95346403378e47d0c8600659456b/studies/templates/studies/study_detail.html#L12

It seems to work without it..?

Comment (okaycj) [reply studies/templates/studies/study_edit.html:27]: Those should be removed as well.  Thanks for the clean up.

Comment (becky-gilbert) [reply studies/templates/studies/study_edit.html]: Done here: 5b882143c2807c71baf6d7aac7bb39399a558cf0 How's that?

**Pull request merged**